### PR TITLE
Feature/detekt r8

### PR DESCRIPTION
**Pull request by lautimfernandez**
## Motivation
Adding Detekt and Ktlint. Migrate to Gradle 5. Enable R8.
Tests were done in andes and mobile android.

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [ ] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [X] I have modified the [Changelog](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CHANGELOG.md) file,
- [ ] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [ ] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [ ] This code adds a new component
- [ ] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

## Check common errors
Whether you are an Android or iOS developer and if you're still there then we'll give you a present:
https://proandroiddev.com/how-to-maximize-androids-ui-reusability-5-common-mistakes-cb2571216a9f

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,13 @@
 # v2.1.0
 ## Added
 - Initial Textfield component
+- - Adding Detekt + Ktlint
```
Comment (juanimoli) [inline CHANGELOG.md:4]:
```suggestion
- Adding Detekt + Ktlint
```

File: config/detekt.yml [hunk 2]
```diff
@@ -0,0 +1,591 @@
+build:
+  maxIssues: 500
```
Comment (juanimoli) [inline config/detekt.yml:2]: 🌝

File: gradle.properties [hunk 3]
```diff
@@ -24,6 +24,11 @@ org.gradle.daemon=true
 org.gradle.caching=true
 ## Max workers for every build. This number can be tweaked
 org.gradle.workers.max=8
+## Disabling R8. TODO: test this properly to enable it
+android.enableR8=true
```
Comment (juanimoli) [inline gradle.properties:28]: no hace falta el flag si seteas `gradleBuildToolsVersion=3.4.2` (que esta mas abajo)

File: config/detekt.yml (hunk 2 above)
Comment (lautimfernandez) [reply config/detekt.yml:2]: Como habiamos hablado, en este pr solo hacemos la migracion, en otro vamos a ir corrigiendo los errores de detekt.

File: gradle.properties (hunk 3 above)
Comment (lautimfernandez) [reply gradle.properties:28]: mas alla de que no haga falta no esta bueno que por las dudas este para que se sepa bien si alguien lo busca? la deje por eso nomas

Comment (juanimoli) [reply gradle.properties:28]: no esta mal, eso ya es como lo quieran dejar ustedes. Ojo con el TODO que dejaste ahi

File: config/detekt.yml (hunk 2 above)
Comment (juanimoli) [reply config/detekt.yml:2]: si, solo queria molestarte

**Pull request merged**